docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,162 @@
+** Pull requests often become a venue to discuss feasibility, underlying technical decisions and design.
+   We are fine with that as well
+** If there is no consensus about the feasibility and implementation,
+   code reviewers are expected to suggest proper channels to maintainers
+* Ensure that the pull request matches the code quality practices: test automation, Javadoc for newly introduced APIs, etc.
+** There is no goal to make the code perfect tho
+* Ensure that the pull requests contain user-friendly changelogs and, if needed, upgrade guidelines
+* Last but not least,
+  guide newcomer contributors who are not familiar with the project processes.
+
+=== Code review non-goals
+
+* Enforce a particular coding style
+** Jenkins core has a complex codebase created by many contributors,
+   and different files have different designs
+** Main goal is 
+* Make contributors to fix unrelated issues
+** Create follow-up issues instead, it is fine to referene them in comments so that the contributor might want to pick them up
+* Breakdown the commit history or squashing PRs
+** Not every contributor is a Git expert, do not request changes in the commit history unless it is necessary
+** Core maintainers can squash PRs during the merge.
+   If you feel this is important, use the `squash-me-please` label
+** We want to keep pull requests atomic when possible (one feature / fix - separate commit),
+   but we can live without it if there is no need to backport changes to the stable baseline.
+
+== Merge process
+
+=== Common merge process
+
+**Step 1: Maintainer checklist**. Merge process can be initiated once a pull request matches the requirements:
+
+* Pull request is compliant wuth requirements to submitters (see the link:/.github/PULL_REQUEST_TEMPLATE.md[pull request template])
+* There are at least 2 approvals for the pull request and no outstanding requests for change
+* Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
+* Changelog entries in the PR title and/or `Proposed changelog entries` are correct
+* Proper changelog labels are set so that the changelog can be generated automatically
+* If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set
+  and there is a `Proposed upgrade guidelines` section in the PR title.
+  (link:https://github.com/jenkinsci/jenkins/pull/4387[example])
+* It would make sense to backport the change to LTS, a Jira issue should exist and be labeled as `lts-candidate`
+
+**Step 2: Marking for merge**. Once the checklist is passed,
+a Core PR Reviewer or a Maintainer can mark the pull request for merge.
+
+* `ready-for-merge` label is set
+* An explicit comment is added to the pull request so that other repository watchers are notified.
+  Example: `Thanks to all contributors! We consider this change as read to be merged towards the next weekly release. It may be merged after a 24hours if there is no negative feedback`
```

Comment:
```suggestion
  Example: `Thanks to all contributors! We consider this change as ready to be merged towards the next weekly release. It may be merged after 24hours if there is no negative feedback`
```